lib/server_health_check.rb

```diff
@@ -25,13 +25,17 @@ def redis!(host: nil, port: 6379)
 
   def active_record!
     begin
-      ActiveRecord::Base.connection
-      @results[:active_record] = OK
-      true
+      if ActiveRecord::Base.connection_pool.with_connection { |connection| connection.active? }
+        @results[:active_record] = OK
+        true
+      else
+        @results[:active_record] = "Failed: unable to connect to database"
+        false
+      end
     rescue NameError
```

Comment:
This gem is designed for creating a health check page. It supports `S3`, `Redis`, and `ActiveRecord` out of the box, but if you don't need or use any of those checks, then you don't actually need to have any of the relevant gems installed (and can still get benefit from the gem). Thus, if AR isn't defined, it probably means you are trying to add the check in a project that doesn't use it, so I think the NameError would be ideal.

As for the suggestion outside the block... I think you already see te problem... I'm not willing to bet that `connection_pool` cannot throw an error (like, misconfigured `database.yml` might throw an exception there perhaps).